### ISSUE TITLE
Setup instructions

### ISSUE THREAD

**Issue by yupinghuang** [semaphoreP/codeastro#12]
1. Is it worthwhile to include installation instructions for [git](https://carpentries.github.io/workshop-template/#git)?
2. The python setup refers to the requirements.txt that technically they get during step 2 (`git clone`).3
3. The diagnostic assignment is in jupyter notebook. Are we expecting them to open it in VSCode or Jupyter (none of IPython or Jupyter is in requirements.txt)?

It might make more sense to organize the setup instruction as follows?
1. Bash/WSL/git setup (or just checking that they have bash and git)
2. Installing miniconda and VSCode
3. Clone this repo and then install the dependencies.
4. Instructions on opening the diagnostics notebook.

Finally, what if they already have a miniconda installation that have python packages in their base environment that might cause issues for this workshop? Is it worth having them create a fresh conda environment? Or should we do it on a case-by-case basis (which might be too late for salvaging broken base environments...)

**Comment by yupinghuang** [comment]
Also, I don't think miniconda installs mkl/libblas/liblapack if you don't install numpy through conda. The pip mkl package is just the python binding so there's a chance the parallelism notebook won't turn out. Maybe asking them to do a conda install mkl-service during conda setup would be helpful (I think this actually removes the need to do mkl from pip https://docs.anaconda.com/mkl-service/)

**Comment by semaphoreP** [comment]
Thanks! I added a link to install git, moved the git clone of this repo to right after git installation, and updated the python installation docs to make a new conda environment and install numpy/jupyter through conda. I also added a brief sentence on launching the diagnostic notebook (let me know if you think more detailed instructions from e.g., a tutorial is warranted). 

Regarding MKL, I added it to the conda install instructions too. I think if they don't have an Intel processor or can't get it configured with numpy, it's not super critical, since we'll only use it for a demo, and the activity doesn't use it.
